cmd/kubelet/app/server.go

```diff
@@ -447,10 +452,30 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 		}
 
 		clientConfig, err := CreateAPIServerClientConfig(s)
+
+		var clientCertificateManager certificate.Manager
 		if err == nil {
+			if utilfeature.DefaultFeatureGate.Enabled(features.RotateKubeletClientCertificate) {
+				nodeName, err := getNodeName(cloud, nodeutil.GetHostname(s.HostnameOverride))
+				if err != nil {
+					return err
+				}
+				clientCertificateManager, err = initializeClientCertificateManager(s.CertDirectory, nodeName, clientConfig.CertData, clientConfig.KeyData)
```

Comment:
If the clientConfig doesn't have certificate data then the certificate manager is initialized without any certs. It will try to use the client to get some as soon `.Start()` is called. However the Certificate Signing Request client is initialized and authenticated is outside the responsibility of the certificate manager. It just uses whatever it is given.

Do you think that if a token is supplied and no certs then the certificate manager should be disabled?